### PR TITLE
Add GQL behave tests for on-disk storage

### PR DESCRIPTION
**Pull request by Darych**
[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: src/query/db_accessor.hpp [hunk 1]
```diff
@@ -459,6 +459,8 @@ class DbAccessor final {
   }
 
   storage::Result<std::optional<VertexAccessor>> RemoveVertex(VertexAccessor *vertex_accessor) {
+    accessor_->PrefetchOutEdges(vertex_accessor->impl_);
```
Comment (Darych) [inline src/query/db_accessor.hpp:462]: The test which failed here now uses `DetachDelete` function.
But we still call this function in `mgp_graph_delete_vertex` without prefetching edges.
Also `DetachDelete` prefetches edges and call `accessor_->DetachDelete()` internally.
Here `accessor_->DeleteVertex()` calls `accessor_->DetachDelete` also.
So we must prefetch edges.

File: tests/gql_behave/run.sh
Comment (as51340) [inline tests/gql_behave/run.sh:1]: I don't think this file should exist. If you need it on pipeline, you can add it as part of the step there. e2e tests also need mgclient but we don't run it by explicitly installing mgclient. Check how e2e tests work :smile:

File: tests/gql_behave/run.py [hunk 2]
```diff
@@ -92,9 +95,27 @@ def add_argument(option, **kwargs):
     behave_args.extend(["--test-suite", parsed_args.test_suite])
     behave_args.extend(["--test-directory", test_directory])
 
+    # Setup database
+    if parsed_args.storage_mode == "in_memory":
```
Comment (as51340) [inline tests/gql_behave/run.py:99]: Add storage mode flag will be merged today -> no need to introduce setup queries, we can just start Memgraph with any storage mode.

**Review by as51340 — CHANGES_REQUESTED** [review]

File: tests/gql_behave/run.sh
Comment (gvolfing) [reply tests/gql_behave/run.sh:1]: Second that.
If it's not a huge chunk of work.

**Review by gvolfing — COMMENTED** [review]
The existence of `run.sh` as @as51340 mentioned. If it's not a huge chunk of work I would rather not have this solved this way.

File: tests/gql_behave/run.sh
Comment (Darych) [reply tests/gql_behave/run.sh:1]: I agree. But there are also in e2e tests:
```
# TODO(gitbuda): Setup mgclient and pymgclient properly.
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:../../libs/mgclient/lib
```

**Comment by Darych** [comment]
In this PR I start to use `tests/ve3` for gql_behave tests. I will remove virtual env from gql_behave directory in separate PR, to not mess everything inside one.

File: tests/setup.sh [hunk 3]
```diff
@@ -59,6 +60,8 @@ if [ $# == 1 ]; then
         OLD_LD_LIBRARY_PATH=$LD_LIBRARY_PATH
         source $toolchain
         NEW_LD_LIBRARY_PATH=$LD_LIBRARY_PATH
+        # TODO(gitbuda): Setup mgclient and pymgclient properly.
+        NEW_LD_LIBRARY_PATH=$NEW_LD_LIBRARY_PATH:"../libs/mgclient/lib"
```
Comment (as51340) [inline tests/setup.sh:64]:
```suggestion
        NEW_LD_LIBRARY_PATH=$LD_LIBRARY_PATH:"../libs/mgclient/lib"
```

**Review by as51340 — APPROVED** [review]

**Review by gvolfing — APPROVED** [review]

**Pull request merged**